src/common/schedulers/slurm_commands.py

```diff
@@ -97,6 +97,25 @@ def update_nodes(
     if we run scontrol update state=fail_state nodeaddr=good_addr nodename=name,
     the scontrol command will fail but nodeaddr will be updated to good_addr.
     """
+    # Only split on , if there is ] before
+    # For ex. "node-[1,3,4-5],node-[20,30]" should split into ["node-[1,3,4-5]","node-[20,30]"]
+    nodes = re.split("(?<=]),", nodes) if type(nodes) is str else nodes
+    nodeaddrs = re.split("(?<=]),", nodeaddrs) if type(nodeaddrs) is str else nodeaddrs
+    nodehostnames = re.split("(?<=]),", nodehostnames) if type(nodehostnames) is str else nodehostnames
+    if nodeaddrs or nodehostnames:
+        # Sorting is only necessary if we set nodeaddrs or nodehostnames
+        nodes_str = ",".join(nodes)
+        sorted_node_names = (
+            check_command_output(f"scontrol show hostlistsorted {nodes_str} | xargs scontrol show hostname", shell=True)
```

Comment:
Let's just put the command on a separate line which will make this easier to read

Reply:
what if this command fails? is the error handled already and the failure reason clear enough?
is there a limit to the number of nodes that these commands can handle?

Reply:
(minor) the official command is `scontrol show hostnames`, but `hostname` works too.